devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp

```diff
@@ -303,75 +303,258 @@ bool WholeBodyDynamicsDevice::openEstimator(os::Searchable& config)
+            if(contactWrenchType.size() == overrideContactFrames.size() && contactWrenchDirection.size() == overrideContactFrames.size() && contactWrenchPosition.size() == overrideContactFrames.size())
+            {
+                yInfo() << "wholeBodyDynamics: parsing the values from the parameters 'overrideContactFrames', 'contactWrenchType', 'contactWrenchDirection' and 'contactWrenchPosition'.";
+            }
+            else
+            {
+                yError() << "wholeBodyDynamics : parameters 'overrideContactFrames', 'contactWrenchType', 'contactWrenchDirection' and 'contactWrenchPosition' sizes are not consistent.";
+                return false;
+            }
+        }
+
+        //create classes "UnknownWrenchContact" for each UnknownWrenchContact
+        wholeBodyDynamics::UnknownWrenchContact wrench[propOverrideContactFrames->size()];
 
-        // If the subModel of the frame still does not have a default contact, we add it
-        if( subModelIndex2DefaultContact[subModelIdx] == iDynTree::FRAME_INVALID_INDEX )
+        for(int ax=0; ax < propOverrideContactFrames->size(); ax++)
         {
-            subModelIndex2DefaultContact[subModelIdx] = defaultContactFramesIdx[i];
+            wrench[ax] = wholeBodyDynamics::UnknownWrenchContact(overrideContactFrames[ax], contactWrenchType[ax], contactWrenchDirection[ax], contactWrenchPosition[ax]);
+
+            if(!wrench[ax].isTypeValid())
+            {
+                yError() << "wholeBodyDynamics : parameter 'contactWrenchType' contains some errors/inconsistencies.";
+                return false;
+            }
         }
-    }
 
+        // We build the overrideContactFramesIdx vector
+        std::vector<iDynTree::FrameIndex> overrideContactFramesIdx;
+        overrideContactFramesIdx.clear();
+        for(size_t i=0; i < overrideContactFrames.size(); i++)
+        {
+            iDynTree::FrameIndex idx = estimator.model().getFrameIndex(overrideContactFrames[i]);
 
-    // Let's check that every submodel has a default contact position
-    for(size_t subModelIdx = 0; subModelIdx < nrOfSubModels; subModelIdx++)
-    {
-        if( subModelIndex2DefaultContact[subModelIdx] == iDynTree::FRAME_INVALID_INDEX )
+            if( idx == iDynTree::FRAME_INVALID_INDEX )
+            {
+                yWarning() << "Frame " << overrideContactFrames[i] << " not found in the model, discarding it";
+            }
+            else
+            {
+                overrideContactFramesIdx.push_back(idx);
+            }
+        }
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// The frames from a single submodel should be no more than 1 if the type is FULL_WRENCH, no more than 3 if the type is PURE_FORCE, and no more than 6 if the type is PURE_FORCE_WITH_KNOWN_DIRECTION.
```

Comment:
You could also add the reference paper in this comment.

Reply:
Done in https://github.com/robotology/whole-body-estimators/pull/48/commits/38c21cdf2a12d33f0fd9a3f736d69353bb89e86e.